validator/fw/string/is_email.go

```diff
@@ -0,0 +1,42 @@
+package string
+
+import (
+	"context"
+	"net/mail"
+
+	"github.com/hashicorp/terraform-plugin-framework-validators/helpers/validatordiag"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+)
+
+// Ensure our implementation satisfies the validator.String interface.
+var _ validator.String = &emailValidator{}
+
+type emailValidator struct{}
+
+func (v emailValidator) Description(_ context.Context) string {
+	return "value must be a valid email address"
+}
+
+func (v emailValidator) MarkdownDescription(ctx context.Context) string {
+	return v.Description(ctx)
+}
+
+func (v emailValidator) ValidateString(ctx context.Context, request validator.StringRequest, response *validator.StringResponse) {
```

Comment:
Just curious: does the `ctx` allow you to do validation basically by doing interdependent field validation? 

Like, wen creating a user, I think, you can't have `admin` and some other field at the same time. But it was damn tricky to prevent this case client side because you had no idea what the other fields were already set to. 

(Side note: I did come up with a clever way to handle that)

Reply:
@chb0github Yes, it's a little more complicated: https://developer.hashicorp.com/terraform/plugin/framework/validation#path-based-attribute-validators

Reply:
But if it is just checking for mutually ex/inclusiveness, there are pre-build validators: https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework-validators@v0.12.0/stringvalidator#AlsoRequires